lib/app.js

```diff
@@ -205,11 +197,55 @@ module.exports = function(tilelive, options) {
             headers["content-encoding"] = headers["content-encoding"] || "gzip";
           }
 
-          res.set(populateHeaders(headers, params, { 200: true }));
-          return res.send(data);
+          return callback(null, data, populateHeaders(headers, params, { 200: true }));
         });
       });
     });
+  };
+
+  app.get(tilePattern, function(req, res, next) {
+    return getTile(req.params.z | 0, req.params.x | 0, req.params.y | 0,
+        (req.params.scale || "@1x").slice(1, 2), req.params.format,
+        function(status, data, headers) {
+          if (status && Number(status) == NaN) {
```

Comment:
As above, please switch to `Error` instances for consistency (with a `status` property). The first argument to a callback should always be an optional error (`null` or `undefined` if no error occurred).

This then becomes something like:

```javascript
if (err && !err.status) {
  return next(err);
}

// ...
```

In writing this, I'm thinking that 404s should be treated as non-errors and detected by checking whether `data == null` (`data === null || data === undefined`), similar to the straightforward 404 check above.